tests/ci/run_minimal_tests.sh

```diff
@@ -23,3 +23,20 @@ build_and_run_minimal_test -DDISABLE_GO=ON -DDISABLE_PERL=ON -DCMAKE_BUILD_TYPE=
 echo "Testing shared lib AWS-LC without Perl/Go in release mode."
 build_and_run_minimal_test -DDISABLE_GO=ON -DDISABLE_PERL=ON -DBUILD_SHARED_LIBS=1 -DCMAKE_BUILD_TYPE=Release
 
+# Special build options
+
+## build option: MY_ASSEMBLER_IS_TOO_OLD_FOR_AVX
+echo "Testing static lib AWS-LC with MY_ASSEMBLER_IS_TOO_OLD_FOR_AVX build option in release mode."
+build_and_run_minimal_test -DCMAKE_BUILD_TYPE=Release -DMY_ASSEMBLER_IS_TOO_OLD_FOR_AVX=ON
+
+echo "Testing shared lib AWS-LC with MY_ASSEMBLER_IS_TOO_OLD_FOR_AVX build option in release mode."
+build_and_run_minimal_test -DBUILD_SHARED_LIBS=1 -DCMAKE_BUILD_TYPE=Release -DMY_ASSEMBLER_IS_TOO_OLD_FOR_AVX=ON
+
+echo "Testing static lib AWS-LC with MY_ASSEMBLER_IS_TOO_OLD_FOR_AVX build option in debug mode."
+build_and_run_minimal_test -DMY_ASSEMBLER_IS_TOO_OLD_FOR_AVX=ON
+
+echo "Testing shared lib AWS-LC with MY_ASSEMBLER_IS_TOO_OLD_FOR_AVX build option in debug mode."
+build_and_run_minimal_test -DBUILD_SHARED_LIBS=1 -DMY_ASSEMBLER_IS_TOO_OLD_FOR_AVX=ON
```

Comment:
Why `build_and_run_minimal_test`? This script is to test the build with `no perl` or `no go`. The build with `-DMY_ASSEMBLER_IS_TOO_OLD_FOR_AVX=ON` still uses `perl` and may use `go`. `tests/ci/run_posix_tests.sh` may be a better place to have these changes.

Reply:
because it is a "minimal test"

Reply:
> Nothing in this file indicates "only for go/perl tests"

This script is developed to test build without go/perl see below references. Previously, `no go/perl` was forced by clients Docker images(replaced recently). This PR is to detect build issues before importing to internal build system. This is why `fips_build_and_test` and `build_and_test` should be used.

1. https://github.com/awslabs/aws-lc/blob/main/BUILDING.md#using-pre-generated-build-files
```
Building without Go now produces a new target, run_minimal_tests in place of run_tests.
```

2. CI BuildSpec https://github.com/awslabs/aws-lc/blob/main/tests/ci/cdk/cdk/codebuild/github_ci_linux_x86_omnibus.yaml#L305-L368

There is 1:1 relationship between buildspec `.yml` and bash file.
```
Build and test aws-lc without Perl/Go.
```